src/main/drivers/rx/rx_sx1280.c

```diff
@@ -0,0 +1,500 @@
+        buf[1] = 0xFF;
+        buf[2] = 0xFF;
+        sx1280WriteCommandBurst(SX1280_RADIO_SET_RX, buf, 3);
+        break;
+
+    case SX1280_MODE_TX:
+        //uses timeout Time-out duration = periodBase * periodBaseCount
+        buf[0] = 0x00; // periodBase = 1ms, page 71 datasheet
+        buf[1] = 0xFF; // no timeout set for now
+        buf[2] = 0xFF; // TODO dynamic timeout based on expected onairtime
+        sx1280WriteCommandBurst(SX1280_RADIO_SET_TX, buf, 3);
+        break;
+
+    case SX1280_MODE_CAD:
+        break;
+
+    default:
+        break;
+    }
+}
+
+void sx1280ConfigLoraModParams(const sx1280_lora_bandwidths_e bw, const sx1280_lora_spreading_factors_e sf, const sx1280_lora_coding_rates_e cr)
+{
+    // Care must therefore be taken to ensure that modulation parameters are set using the command
+    // SetModulationParam() only after defining the packet type SetPacketType() to be used
+
+    uint8_t rfparams[3] = {0};
+
+    rfparams[0] = (uint8_t)sf;
+    rfparams[1] = (uint8_t)bw;
+    rfparams[2] = (uint8_t)cr;
+
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_MODULATIONPARAMS, rfparams, 3);
+
+    switch (sf)
+    {
+    case SX1280_LORA_SF5:
+    case SX1280_LORA_SF6:
+        sx1280WriteRegister(0x925, 0x1E); // for SF5 or SF6
+        break;
+    case SX1280_LORA_SF7:
+    case SX1280_LORA_SF8:
+        sx1280WriteRegister(0x925, 0x37); // for SF7 or SF8
+        break;
+    default:
+        sx1280WriteRegister(0x925, 0x32); // for SF9, SF10, SF11, SF12
+    }
+}
+
+void sx1280SetFrequencyHZ(const uint32_t reqFreq)
+{
+    uint8_t buf[3] = {0};
+
+    uint32_t freq = (uint32_t)((double)reqFreq / (double)SX1280_FREQ_STEP);
+    buf[0] = (uint8_t)((freq >> 16) & 0xFF);
+    buf[1] = (uint8_t)((freq >> 8) & 0xFF);
+    buf[2] = (uint8_t)(freq & 0xFF);
+
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_RFFREQUENCY, buf, 3);
+}
+
+void sx1280SetFrequencyReg(const uint32_t freq)
+{
+    uint8_t buf[3] = {0};
+
+    buf[0] = (uint8_t)((freq >> 16) & 0xFF);
+    buf[1] = (uint8_t)((freq >> 8) & 0xFF);
+    buf[2] = (uint8_t)(freq & 0xFF);
+
+    sx1280WriteCommandBurst(SX1280_RADIO_SET_RFFREQUENCY, buf, 3);
+}
+
+int32_t sx1280GetFrequencyError(void)
+{
+    uint8_t efeRaw[3] = {0};
+    uint32_t efe = 0;
+    double efeHz = 0.0;
+
+    efeRaw[0] = sx1280ReadRegister(SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB);
+    efeRaw[1] = sx1280ReadRegister(SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB + 1);
+    efeRaw[2] = sx1280ReadRegister(SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MSB + 2);
+    efe = (efeRaw[0] << 16) | (efeRaw[1] << 8) | efeRaw[2];
+
+    efe &= SX1280_REG_LR_ESTIMATED_FREQUENCY_ERROR_MASK;
+
+    //efeHz = 1.55 * (double)complement2(efe, 20) / (1600.0 / (double)GetLoRaBandwidth() * 1000.0);
+    return efeHz;
+}
+
+void sx1280AdjustFrequency(int32_t offset, const uint32_t freq)
+{
+    //Not implemented
+    UNUSED(offset);
+    UNUSED(freq);
+}
```

Comment:
Maybe this should be removed too? Can be added back if/when needed?

Reply:
Here it is just a stub to make frequency adjust code consistent between sx1280 and sx127x. I will adjust the comment to reflect the intention better.

Reply:
Changed comment to reflect